src/test/scala/com/cognite/spark/datasource/SparkTest.scala

```diff
@@ -29,4 +32,17 @@ trait SparkTest extends CdpConnector {
 
     (modelId, revisionId)
   }
+
+  def retryUntil[A](action: => A, shouldRetry: A => Boolean): A =
```

Comment:
I think the name was my idea... but it should be `retryWhile` because that's what it actually does. ;)